grails-app/controllers/org/pih/warehouse/inventory/StockTransferController.groovy

```diff
@@ -122,4 +125,19 @@ class StockTransferController {
         }
 
     }
+
+    def rollback = {
+        Order order = Order.get(params.id)
+        if (!order) {
+            throw new IllegalArgumentException("Can't find order with given id: ${params.id}")
+        }
+        try {
+            shipmentService.rollbackLastEvent(order.shipments.last() as Shipment)
```

Comment:
Make it null save and take the first shipment `order.shipments?.first()` and add above this line the comment: `
        // For returns there should be only one shipment for the given order`